### PR TITLE
Issue #3476: Improve escaped unicode chars detection for AvoidEscaped…

### PR DESCRIPTION
**Pull request by Dziman**
Issue #3476: Wrong warning from AvoidEscapedUnicodeCharacters

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java [hunk 1]
```diff
@@ -242,7 +242,10 @@ public void visitToken(DetailAST ast) {
      * @return true if literal has Unicode chars.
      */
     private static boolean hasUnicodeChar(String literal) {
-        return UNICODE_REGEXP.matcher(literal).find();
+        final Pattern escapedBackslashPattern = Pattern.compile("\\\\\\\\");
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java]: please do compile and save it in final field of class, to improve performance a bit.

Comment (Dziman) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java]: @romani fixed

**Comment by romani** [comment]
code is good.

we need to do regression testing over guava report[, as this Check was used in google config
http://checkstyle.sourceforge.net/google_style.html
http://checkstyle.sourceforge.net/reports/google-style/guava/
to do this, please use following tools:
https://github.com/checkstyle/contribution/tree/master/checkstyle-tester
https://github.com/checkstyle/contribution/tree/master/patch-diff-report-tool
please run them and share results in github.io . All instructions are at README.md files.

**Comment by Dziman** [comment]
@romani 
Diff: https://dziman.github.io/checkstyle-3476/diff/
Report: https://dziman.github.io/checkstyle-3476/

**Comment by romani** [comment]
code is merged.
thanks a lot for your contribution! You are welcome for any other contributions.

**Pull request merged**